### PR TITLE
[Chore] Search Updates

### PR DESCRIPTION
**Pull request by grahamhency**
## PR Checklist
Please check if your PR fulfills the following requirements:
- [x] The commit message follows our guidelines: https://github.com/mobi/goponents/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
- [x] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?
The search bar in the middle section of the header that threw an error in a compiled application.

## What is the new behavior?
Fixed this issue, as well as implemented a few other adjustments and updates to the search.

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/go-lib/src/lib/components/go-header/go-header.component.ts [hunk 1]
```diff
@@ -72,10 +70,6 @@ export class GoHeaderComponent implements OnChanges, OnDestroy {
     this.sideNavService.toggleNav();
   }
 
-  middleContentExists(): boolean {
-    return this.middleSection && this.middleSection.nativeElement.childElementCount > 0;
-  }
```
Comment (grahamhency) [inline projects/go-lib/src/lib/components/go-header/go-header.component.ts:77]: This is totally unnecessary and was causing issues, thus I am removing it.

File: projects/go-lib/src/lib/components/go-search/go-search.component.ts [hunk 2]
```diff
@@ -75,6 +80,18 @@ export class GoSearchComponent implements OnInit {
...
+    if (event.code === 'ArrowDown') {
+      this.goSearchService.arrowDownEvent.emit();
+    }
+  }
+
+  focusOnSearch(): void {
+    this.goSearchService.focusOnSearchEvent.subscribe(() => {
+      this.searchInput.nativeElement.focus();
+    });
+  }
```
Comment (StevenUlmer) [inline projects/go-lib/src/lib/components/go-search/go-search.component.ts:93]: What was the purpose behind adding this here? Was it just to apply the correct css?

File: projects/go-lib/src/lib/components/go-search/go-search.component.ts [hunk 3]
```diff
@@ -75,6 +80,18 @@ export class GoSearchComponent implements OnInit {
     }
   }
 
+  keyDown(event: Partial<KeyboardEvent>): void {
+    if (event.code === 'ArrowDown') {
+      this.goSearchService.arrowDownEvent.emit();
+    }
+  }
```
Comment (StevenUlmer) [inline projects/go-lib/src/lib/components/go-search/go-search.component.ts:87]: Should we allow for more keyboard navigation than just arrow key down here? If we're looking for navigation within search results the user should have the ability to go down or up in the results right? Should we also add the ability to close the search with the escape key? Not sure if that works currently, but I would assume it doesn't.

**Comment by grahamhency** [comment]
I believe there is a misunderstanding on how this component is implemented. This component does not handle rendering the search results because formatting could differ vastly between products and implementations.

That being said, the reasoning for adding these keyboard and focus events are so that you can handle the navigation from your results component back into the search component.

`arrowDownEvent` is so that you know when the user navigates down, out of the search input.

`focusOnSearchInput` so for then you need to apply focus back on the search input.

The actual navigation through the results will have to be handled upon on the implementation of this component. Does all of this make sense?

File: projects/go-lib/src/lib/components/go-search/go-search.component.ts (hunk 2 above)
Comment (grahamhency) [reply projects/go-lib/src/lib/components/go-search/go-search.component.ts:93]: See my comment below.

File: projects/go-lib/src/lib/components/go-search/go-search.component.ts (hunk 3 above)
Comment (grahamhency) [reply projects/go-lib/src/lib/components/go-search/go-search.component.ts:87]: See my comment below, but we could look at doing the escape key as well.

**Pull request merged**